uiowahjmjohnsonselt2024/projectdirectory-selt_2024_team_009
# Pull Request: ntegrate Server Creation Flow and Display Opponent Details

## What does this PR do?: 
This PR integrates the flow between server creation and the game grid display while also providing a dedicated view for opponent details. It ensures that players can view the necessary information about their opponents (such as treasures, items, remaining AP points, and cells occupied) to strategize their moves. Additionally, the server creation process seamlessly transitions to the game grid display, and the grid updates in real-time as players make moves.

## Related Issues: 
- Closes #118   Display Opponent Details 
- Closes #120 Integration of Server Creation and Game Grid Display

## Testing
1. Create a server and start a new game.
2. Verify that the game grid is displayed immediately after the server is created.
3. Check that the game grid updates in real-time as players make moves.
4. Ensure the player can view details for all opponents, including treasures, items, remaining AP points, and cells occupied.
5. Test that this information updates dynamically as the game progresses.
6. Verify that the view is accessible during the player’s turn and shows data for all opponents in the game.
7. Confirm that error handling is in place for server creation and grid display failures.

## Checklist

## Extra information (if needed)

This issue closes the following:

- **Display Opponent Details #118**: A dedicated view or section displays the following details for each opponent:
  - Treasures in their inventory
  - Items in their inventory
  - Remaining AP points
  - Number of cells occupied

- **Integration of Server Creation and Game Grid Display #120**: Ensures the server creation and game grid flow are integrated, with real-time updates and seamless transitions between creating a server and viewing the game grid.
